features/include/pcl/features/gasd.h

```diff
@@ -0,0 +1,252 @@
+      };
+
+      /** \brief Empty constructor. */
+      GASDEstimation () :
+          view_direction_ (0.0f, 0.0f, 1.0f),
+          shape_half_grid_size_ (3),
+          shape_hists_size_ (1),
+          color_half_grid_size_ (2),
+          color_hists_size_ (12),
+          use_color_ (true),
+          shape_interp_ (INTERP_NONE),
+          color_interp_ (INTERP_NONE)
+      {
+        search_radius_ = 0;
+        k_ = 1;
+        feature_name_ = "GASDEstimation";
+      }
+
+      /** \brief Set the view direction.
+       * \param[in] dir view direction
+       */
+      inline void
+      setViewDirection (Eigen::Vector3f &dir)
+      {
+        view_direction_ = dir;
+      }
+
+      /** \brief Set the shape half grid size.
+       * \param[in] shgs shape half grid size
+       */
+      inline void
+      setShapeHalfGridSize (int shgs)
+      {
+        shape_half_grid_size_ = shgs;
+      }
+
+      /** \brief Set the shape histograms size. If size is 1, then each histogram bin will store the number
+       * of points that belong to its correspondent cell in the 3D regular grid. If size > 1, then for each cell
+       * it will be computed a histogram of normalized distances between each sample and the cloud centroid
+       * \param[in] shs shape histograms size
+       */
+      inline void
+      setShapeHistsSize (int shs)
+      {
+        shape_hists_size_ = shs;
+      }
+
+      /** \brief Set the color half grid size.
+       * \param[in] chgs color half grid size
+       */
+      inline void
+      setColorHalfGridSize (int chgs)
+      {
+        color_half_grid_size_ = chgs;
+      }
+
+      /** \brief Set the color histograms size (number of bins in the hue histogram for each cell of the 3D regular grid).
+       * \param[in] chs color histograms size
+       */
+      inline void
+      setColorHistsSize (int chs)
+      {
+        color_hists_size_ = chs;
+      }
+
+      /** \brief Set if color should be used.
+       * \param[in] use if true, color is used
+       */
+      inline void
+      setUseColor (bool use)
+      {
+        use_color_ = use;
+      }
+
+      /** \brief Set the shape histograms interpolation method.
+       * \param[in] interp shape histograms interpolation method
+       */
+      inline void
+      setShapeHistsInterpMethod (HistogramInterpolationMethod interp)
+      {
+        shape_interp_ = interp;
+      }
+
+      /** \brief Set the color histograms interpolation method.
+       * \param[in] interp color histograms interpolation method
+       */
+      inline void
+      setColorHistsInterpMethod (HistogramInterpolationMethod interp)
+      {
+        color_interp_ = interp;
+      }
+
+      /** \brief Returns the transformation aligning the point cloud to the canonical coordinate system
+       * \param[out] trans transformation
+       */
+      void
+      getTransform (Eigen::Matrix4f &trans)
+      {
+        trans = transform_;
+      }
```

Comment:
You should change this method to take no arguments and return a const ref to transform_.